test/unit/spec/util/jobLogViewerSpec.js

```diff
@@ -48,217 +222,493 @@ define(['util/jobLogViewer', 'common/runtime'], (JobLogViewer, Runtime) => {
             const arg = {
-                node: hostNode,
-                jobId: 'someFakeJob',
+                node: this.node,
+                jobId: 'test_job_start',
             };
-            await viewer.start(arg);
-            expect(hostNode.querySelector('div[data-element="kb-log"]')).toBeDefined();
-            viewer.detach();
-            expect(hostNode.innerHTML).toBe('');
+            await this.jobLogViewerInstance.start(arg);
+            expect(this.node.querySelector('div[data-element="status-line"]')).toBeDefined();
+            this.jobLogViewerInstance.detach();
+            expect(this.node.innerHTML).toBe('');
         });
 
-        it('Should send a bus messages requesting job status information at startup', async (done) => {
-            const viewer = JobLogViewer.make();
-            const jobId = 'testJob1';
+        it('Should send bus messages requesting job status information at startup', async function () {
+            const jobId = 'test_bus_request';
             const arg = {
-                node: hostNode,
+                node: this.node,
                 jobId: jobId,
             };
-            runtimeBus.on('request-job-status', (msg) => {
-                expect(msg).toEqual({ jobId: jobId });
-                viewer.detach();
-                done();
+            await this.jobLogViewerInstance.start(arg).then(() => {
+                return new Promise((resolve) => {
+                    this.runtimeBus.on('request-job-status', (msg) => {
+                        expect(msg).toEqual({ jobId: jobId });
+                        resolve();
+                    });
+                });
             });
-            await viewer.start(arg);
         });
 
-        it('Should react to job status messages', async (done) => {
-            const viewer = JobLogViewer.make();
-            const jobId = 'testJobStatusMsg';
+        it('Should start with all buttons disabled', async function () {
             const arg = {
-                node: hostNode,
-                jobId: jobId,
+                node: this.node,
+                jobId: 'testBtnState',
             };
-            runtimeBus.on('request-job-status', (msg) => {
-                expect(msg).toEqual({ jobId: jobId });
-                runtimeBus.send(
-                    {
-                        jobId: jobId,
-                        jobState: {
-                            status: 'running',
-                        },
-                    },
-                    {
-                        channel: {
+            await this.jobLogViewerInstance.start(arg);
+            const btns = this.node.querySelectorAll('div[data-element="header"] button');
+            btns.forEach((btn) => {
+                expect(btn).toHaveClass('disabled');
+            });
+        });
+
+        describe('initial widget state', () => {
+            describe('should be determining the state with no job object', () => {
```

Comment:
The status line reads 'Determining job state...' if there is no job state object supplied on widget start.